Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,243 @@
+                    nextBalance = float.TryParse(userHistory[i + 1], out float nextResult) ? float.Parse(userHistory[i + 1]) : float.NaN;
+
+                    biggestLoss = ((nextBalance - balance) < biggestLoss) ? (nextBalance - balance) : biggestLoss;
+
+
+
+                    if (biggestLoss == nextBalance - balance && biggestLoser != currentBalanceHolder)
+                    {
+                        switch (holderCount)
+                        {
+                            case 0:
+                                {
+                                    biggestLoser = currentBalanceHolder;
+                                    holderCount++;
+                                    break;
+                                }
+
+                            case 1:
+                                {
+                                    biggestLoser += ($", {currentBalanceHolder}");
+                                    holderCount++;
+                                    break;
+                                }
+                            case 3:
+                                {
+                                    biggestLoser += ($" and {currentBalanceHolder}");
+                                    break;
+                                }
+                        }
+                    }
+                }
+            }
+            string stringBiggestLoss = Math.Abs(biggestLoss).ToString();
+            return ($"{biggestLoser} lost the most money. -¤{stringBiggestLoss}.");
         }
 
         /// <summary>
         /// Return name and current money of the richest person.
         /// </summary>
         public static string FindRichestPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0)
+            {
+                return "N/A.";
+            }
+            int holderCount = 0;
+            float highestLastBalance = float.MinValue;
+            string currentBalanceHolder;
+            string highestLastBalanceHolder = "";
+            foreach (string entry in peopleAndBalances)
+            {
+                string[] userHistory = entry.Split(',');
+
+                currentBalanceHolder = userHistory[0];
+                    
+                float currentBalanceHolderFinalBalance = float.Parse(userHistory[^1]);
+
+                if (highestLastBalance < currentBalanceHolderFinalBalance)
+                {
+                    highestLastBalanceHolder = currentBalanceHolder;
+                    holderCount = 1;
+                }
+
+                if (highestLastBalance == currentBalanceHolderFinalBalance && highestLastBalanceHolder != currentBalanceHolder)
+                {                   
+                        switch (holderCount)
+                        {
+                            case 1:
+                                {
+                                    highestLastBalanceHolder += ($", {currentBalanceHolder}");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use a StringBuilder instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4953253)